elytron/src/main/java/org/wildfly/extension/elytron/JaasRealmDefinition.java

```diff
@@ -0,0 +1,156 @@
+                    .setAttributeGroup(ElytronDescriptionConstants.FILE)
+                    .setRestartAllServices()
+                    .build();
+
+    static final SimpleAttributeDefinition MODULE = new SimpleAttributeDefinitionBuilder(ClassLoadingAttributeDefinitions.MODULE)
+            .setRequired(false)
+            .setRestartAllServices()
+            .build();
+
+    private static final SimpleAttributeDefinition ENTRY = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ENTRY, ModelType.STRING, false)
+            .setRequired(true)
+            .setRestartAllServices()
+            .build();
+
+    private static final SimpleAttributeDefinition CALLBACK_HANDLER = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.CALLBACK_HANDLER, ModelType.STRING, false)
+            .setRequired(false)
+            .setRestartAllServices()
+            .build();
+
+    static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[]{JAAS_CONFIG_PATH, RELATIVE_TO, ENTRY, MODULE, CALLBACK_HANDLER};
+
+    private static final AbstractAddStepHandler ADD = new JaasRealmDefinition.RealmAddHandler();
+    private static final OperationStepHandler REMOVE = new TrivialCapabilityServiceRemoveHandler(ADD, SECURITY_REALM_RUNTIME_CAPABILITY);
+
+    JaasRealmDefinition() {
+        super(new Parameters(PathElement.pathElement(ElytronDescriptionConstants.JAAS_REALM), ElytronExtension.getResourceDescriptionResolver(ElytronDescriptionConstants.JAAS_REALM))
+                .setAddHandler(ADD)
+                .setRemoveHandler(REMOVE)
+                .setAddRestartLevel(OperationEntry.Flag.RESTART_RESOURCE_SERVICES)
+                .setRemoveRestartLevel(OperationEntry.Flag.RESTART_RESOURCE_SERVICES)
+                .setCapabilities(SECURITY_REALM_RUNTIME_CAPABILITY));
+    }
+
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        AbstractWriteAttributeHandler write = new ElytronReloadRequiredWriteAttributeHandler(ATTRIBUTES);
+        for (AttributeDefinition current : ATTRIBUTES) {
+            resourceRegistration.registerReadWriteAttribute(current, null, write);
+        }
+    }
+
+    private static class RealmAddHandler extends BaseAddHandler {
+
+        private RealmAddHandler() {
+            super(SECURITY_REALM_RUNTIME_CAPABILITY, ATTRIBUTES);
+        }
+
+        @Override
+        protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+            ServiceTarget serviceTarget = context.getServiceTarget();
+            RuntimeCapability<Void> runtimeCapability = SECURITY_REALM_RUNTIME_CAPABILITY.fromBaseCapability(context.getCurrentAddressValue());
+            String jaasConfigPath = JAAS_CONFIG_PATH.resolveModelAttribute(context, model).asString();
```

Comment:
Is this attribute required? Looks like that needs to be specified in `JAAS_CONFIG_PATH`.

Reply:
not required, fixed